lib/lists/indexing_single_field.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:screensite/state/generic_state_notifier.dart';
+import 'dart:async';
+
+//ignore: must_be_immutable
+class IndexingSingleField extends ConsumerWidget {
+  final QueryDocumentSnapshot<Map<String, dynamic>> document;
+  final StateNotifierProvider<GenericStateNotifier<Map<String, bool>>,
+      Map<String, bool>> editings;
+  final Map<String, TextSelection> textSelections;
+  final TextEditingController textEditingController = TextEditingController();
+  Timer? timer;
+
+  IndexingSingleField(this.document, this.editings, this.textSelections);
+
+  void onChanged(String value) {
```

Comment:
I thought we already had a common widget for that, pls have a look... You might have ideas how to improve that one...